Sources/GRPCHTTP2TransportNIOTransportServices/HTTP2ClientTransport+TransportServices.swift

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(Network)
+public import GRPCCore
+public import GRPCHTTP2Core
+public import NIOTransportServices  // has to be public because of default argument value in init
+public import NIOCore  // has to be public because of EventLoopGroup param in init
+
+private import Network
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+extension HTTP2ClientTransport {
+  /// A ``GRPCCore/ClientTransport`` using HTTP/2 built on top of `NIOTransportServices`.
+  ///
+  /// This transport builds on top of SwiftNIO's Transport Services networking layer and is the recommended
+  /// variant for use for use on Darwin-based platforms **only** (macOS, iOS, etc.).
```

Comment:
It would be hard to recommend it anywhere else since it won't exist on other platforms 😃

Reply:
Haha yeah my point with the only was to make it clear it's only available on Darwin platforms, but maybe it doesn't clarify as I expected it to. I'll get rid of it.